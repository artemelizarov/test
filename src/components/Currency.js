import React from "react";

function Currency(props) {
  function onClickFavorite(currency) {
    return () => {
      props.onChangeFavorite(currency);
    };
  }

  return (
    <li className="Currency">
      <span>
        {props.code} - {props.name}
      </span>
      {props.rate ? <span> {props.rate}</span> : null}
      <button
        style={{
          padding: 0,
          border: "none",
          font: "inherit",
          color: "inherit",
          backgroundColor: "transparent",
          outline: "none"
        }}
        onClick={onClickFavorite(props.code)}
      >
        {props.favorite ? <span>&#9733;</span> : <span>&#9734;</span>}
      </button>
    </li>
  );
}

export default Currency;
