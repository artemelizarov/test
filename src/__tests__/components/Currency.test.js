import Currency from "../../components/Currency";

describe("render Currency component", () => {
  let wrapper1, wrapper2;

  beforeAll(() => {
    wrapper1 = mount(
      <Currency code="USD" name="United States dollar" rate={55.4} />
    );
    wrapper2 = mount(<Currency code="USD" name="United States dollar" />);
  });

  test("render Currency component", () => {
    expect(wrapper1.find(".Currency").length).toBe(1);
  });

  test("render Currency component whith rate", () => {
    expect(wrapper1.find(".Currency").instance().textContent).toEqual(
      "USD - United States dollar 55.4☆"
    );
  });

  test("render Currency component whithout rate", () => {
    expect(wrapper2.find(".Currency").instance().textContent).toEqual(
      "USD - United States dollar☆"
    );
  });
});

describe("snapshot-test Currency component", () => {
  test("Renders correct properties", () => {
    shallowExpect(
      <Currency code="USD" name="United States dollar" rate={55.4} />
    ).toMatchSnapshot();
  });
});
