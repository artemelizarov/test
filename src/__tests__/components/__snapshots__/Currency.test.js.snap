// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot-test Currency component Renders correct properties 1`] = `
<li
  className="Currency"
>
  <span>
    USD
     - 
    United States dollar
  </span>
  <span>
     
    55.4
  </span>
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "border": "none",
        "color": "inherit",
        "font": "inherit",
        "outline": "none",
        "padding": 0,
      }
    }
  >
    <span>
      â˜†
    </span>
  </button>
</li>
`;
